src/azul/service/avro_pfb.py

```diff
@@ -0,0 +1,463 @@
+                             files_entity: bool = False) -> Iterable[JSON]:
+    for entity_type, field_type in field_type.items():
+        plural = isinstance(field_type, list)
+        if plural:
+            field_type = one(field_type)
+        if isinstance(field_type, dict):
+            yield {
+                "name": entity_type,
+                "type": {
+                    # This is always an array, even if singleton is passed in
+                    "type": "array",
+                    "items": {
+                        "name": entity_type,
+                        "type": "record",
+                        "fields": list(_entity_schema_recursive(field_type,
+                                                                files_entity=files_entity))
+                    }
+                }
+            }
+        elif field_type in _nullable_to_pfb_types:
+            # Exceptions are fields that do not become lists during aggregation
+            exceptions = ('donor_count', 'total_estimated_cells')
+            if files_entity and not plural or entity_type in exceptions:
+                yield {
+                    "name": entity_type,
+                    "type": list(_nullable_to_pfb_types[field_type]),
+                }
+            else:
+                yield {
+                    "name": entity_type,
+                    "type": {
+                        "type": "array",
+                        "items": list(_nullable_to_pfb_types[field_type]),
+                    }
+                }
+        elif field_type is pass_thru_uuid4:
+            yield {
+                "name": entity_type,
+                "default": None,
+                "type": ["string"],
+                "logicalType": "UUID"
+            }
+        elif field_type is value_and_unit:
+            yield {
+                "name": entity_type,
+                "type": {
+                    "name": entity_type,
+                    "type": "array",
+                    "items": [
+                        # FIXME: Change 'string' to 'null' in https://github.com/DataBiosphere/azul/issues/2462
+                        "string",
+                        {
+                            "name": entity_type,
+                            "type": "record",
+                            "fields": [
+                                {
+                                    "name": name,
+                                    # Although, not technically a null_str, it's effectively the same
+                                    "type": _nullable_to_pfb_types[null_str]
+                                } for name in ('value', 'unit')
+                            ]
+                        }
+                    ]
+                }
+            }
+        elif field_type in (pass_thru_json, pass_thru_int):
+            # Pass thru types are used only for aggregation and are excluded
+            # from actual hits
+            continue
```

Comment:
`pass` is preferred

https://github.com/DataBiosphere/azul/blob/develop/CONTRIBUTING.rst#118control-flow

here and wherever